JSONAdapterInterface/IJSONValue.h

```diff
@@ -80,6 +90,29 @@ namespace systelab { namespace json {
 		// Factory methods
 		virtual std::unique_ptr<IJSONValue> buildValue(Type) const = 0;
 		virtual std::unique_ptr<IJSONDocument> buildDocument() const = 0;
+
+	private:
+
+		// Array iterator for range-loop
+		template <typename BaseIterator_, typename Ref_>
+		class ArrayIterator_ : public BaseIterator_
+		{
+		public:
+
+			virtual ~ArrayIterator_() = default;
```

Comment:
not needed as this is a final class, adding this virtual method implies allocating vtable unnecessarily